src/sail_lean_backend/pretty_print_lean.ml

```diff
@@ -0,0 +1,115 @@
+open Libsail
+
+open Type_check
+open Ast
+open Ast_defs
+open Ast_util
+open Reporting
+open Rewriter
+open PPrint
+open Pretty_print_common
+
+let is_enum env id = match Env.lookup_id id env with Enum _ -> true | _ -> false
+
+let pat_is_plain_binder env (P_aux (p, _)) =
+  match p with
+  | (P_id id | P_typ (_, P_aux (P_id id, _))) when not (is_enum env id) -> Some (Some id)
+  | P_wild | P_typ (_, P_aux (P_wild, _)) -> Some None
+  | P_var (_, _) -> Some (Some (Id_aux (Id "var", Unknown)))
+  | P_app (_, _) -> Some (Some (Id_aux (Id "app", Unknown)))
+  | P_vector _ -> Some (Some (Id_aux (Id "vect", Unknown)))
+  | P_tuple _ -> Some (Some (Id_aux (Id "tuple", Unknown)))
+  | P_list _ -> Some (Some (Id_aux (Id "list", Unknown)))
+  | P_cons (_, _) -> Some (Some (Id_aux (Id "cons", Unknown)))
+  | P_lit _ -> Some (Some (Id_aux (Id "lit", Unknown)))
+  | _ -> None
+
+(* Copied from the Coq PP *)
+let args_of_typ l env typs =
+  let arg i typ =
+    let id = mk_id ("arg" ^ string_of_int i) in
+    ((P_aux (P_id id, (l, mk_tannot env typ)), typ), E_aux (E_id id, (l, mk_tannot env typ)))
+  in
+  List.split (List.mapi arg typs)
+
+(* Copied from the Coq PP *)
+let rec untuple_args_pat typs (P_aux (paux, ((l, _) as annot)) as pat) =
```

Comment:
It would probably be a good idea to keep the comment from the Coq version here.  At some point we should start pulling out some functions that are common to both backends and put them into Sail's core.

Reply:
What would be a good file to place those functions common to both exporters?